src/cephadm/cephadm

```diff
@@ -3439,6 +3453,186 @@ class CephContainer:
                                 desc=self.entrypoint, timeout=timeout)
         return out
 
+
+#####################################
+
+class MgrListener(Thread):
+    def __init__(self, agent: 'CephadmAgent') -> None:
+        self.agent = agent
+        self.stop = False
+        super(MgrListener, self).__init__(target=self.run)
+
+    def run(self) -> None:
+        listenSocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        listenSocket.bind(('0.0.0.0', self.agent.listener_port))
```

Comment:
Maybe we could specify a starting port (~8000, like ceph daemons?) and try bind in a loop until we find a free one.  Then the port the first time we report in to the mgr.  Otherwise agents from multiple clusters would collide with each other (and/or some other random competing service).

Reply:
how is the host firewall configured for this? would we need to specify a range of potential ports where the cephadm endpoint binds? What happens at mgr failover time, could the port change forcing a reconfig of all cephadm agents?

Reply:
Now setup to check the next 1000 ports following the port specified in the mgr config option. Might need to look into firewall stuff, but a lot of our other daemons are doing similar.